Error when terminating instance with attached network interface

I attached a network interface to this instance manually.
This caused `gravitonctl termiante` to fail because the network interface is dependent on the instance.

There should be a check in place to see if the instance has any dependent resources

```
mathis@MacBook-Pro ~ % gravitonctl terminate instance
INFO[0000] Terminating: i-0546d514990d2711f
panic: DependencyViolation: resource sg-0e49b3c51acab7a5a has a dependent object
	status code: 400, request id: 796b5740-b055-4ad9-9aed-b950c8493544

goroutine 1 [running]:
gravitonctl/cmd.glob..func5(0x105334400, {0x1400006def0, 0x1, 0x1})
	/Users/mathis/git/gravitonctl/cmd/terminate.go:26 +0xd8
github.com/spf13/cobra.(*Command).execute(0x105334400, {0x1400006dec0, 0x1, 0x1})
	/Users/mathis/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:860 +0x640
github.com/spf13/cobra.(*Command).ExecuteC(0x14000330000)
	/Users/mathis/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:974 +0x410
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/mathis/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:902
gravitonctl/cmd.Execute()
	/Users/mathis/git/gravitonctl/cmd/root.go:21 +0xcc
main.main()
	/Users/mathis/git/gravitonctl/main.go:12 +0x20
```

--- a/pkg/aws/terminate.go
+++ b/pkg/aws/terminate.go
@@ -13,6 +13,11 @@ func Terminate(name string) error {
 		return err
 	}
 
+	if len(instances) == 0 {
+		log.Infof("No instance with name %s found", name)
+		return nil
+	}
+
 	for _, instance := range instances {
 		if *instance.State.Name == ec2.InstanceStateNameTerminated {
 			continue
@@ -20,15 +25,14 @@ func Terminate(name string) error {
 			//TODO this needs work!
 		}
 
-		log.Infof("Terminating: %s", *instance.InstanceId)
-
 		// range over and delete network interfaces
-		for niNum, ni := range instance.NetworkInterfaces {
-			if niNum == 0 {
+		for _, ni := range instance.NetworkInterfaces {
+			if *ni.Attachment.DeviceIndex == 0 {
+				// root network interface, cannot detach these
 				continue
 			}
 
-			log.Infof("Deleting ni %s", *ni.NetworkInterfaceId)
+			log.Infof("Detaching network interface: %s", *ni.NetworkInterfaceId)
 
 			_, err := ec2svc.DetachNetworkInterface(&ec2.DetachNetworkInterfaceInput{
 				AttachmentId: ni.Attachment.AttachmentId,
@@ -40,19 +44,24 @@ func Terminate(name string) error {
 
 			err = ec2svc.WaitUntilNetworkInterfaceAvailable(&ec2.DescribeNetworkInterfacesInput{
 				NetworkInterfaceIds: []*string{
-					instance.InstanceId,
+					ni.NetworkInterfaceId,
 				},
 			})
 			if err != nil {
 				return err
 			}
 
-			_, err = ec2svc.DeleteNetworkInterface(&ec2.DeleteNetworkInterfaceInput{
-				NetworkInterfaceId: ni.NetworkInterfaceId,
-			})
-			if err != nil {
-				return err
-			}
+			log.Warnf("Network interface: %s detached, but not deleted!", *ni.NetworkInterfaceId)
+
+			// not deleting network interfaces as they may have elastic IP's attached
+			// checking for, and deleting these elastic IP's will be implemented later
+
+			// _, err = ec2svc.DeleteNetworkInterface(&ec2.DeleteNetworkInterfaceInput{
+			// 	NetworkInterfaceId: ni.NetworkInterfaceId,
+			// })
+			// if err != nil {
+			// 	return err
+			// }
 		}
 
 		terminateInstancesInput := &ec2.TerminateInstancesInput{
@@ -60,6 +69,8 @@ func Terminate(name string) error {
 				instance.InstanceId,
 			},
 		}
+
+		log.Infof("Terminating: %s", *instance.InstanceId)
 
 		_, err = ec2svc.TerminateInstances(terminateInstancesInput)
 
@@ -88,11 +99,6 @@ func Terminate(name string) error {
 				return err
 			}
 		}
-	}
-
-	if len(instances) == 0 {
-		log.Infof("No instance with name %s found", name)
-		return nil
 	}
 
 	return err

